### PR TITLE
make 0.3 more real

### PR DESCRIPTION
**Pull request by jholman**

### CODE REVIEW

File: src/app.js [hunk 1]
```diff
@@ -33,22 +34,7 @@ class App extends React.Component {
...
-          clickAction: () => this.changeModalTo(333),
-          buttonLabel: 'A church' // We probably don't actually have a church
-        },
-        { id: 9000, pos: { x: 860, y: 440 }, color: '#BB3399', icon: '☢️',
-          clickAction: () => this.changeModalTo(9000),
-          buttonLabel: 'Mine',
-          modal: MineModal,
-        },
-      ],
+      economy: new Economy(),
```
Comment (garrettgsb) [inline src/app.js]: So goooood 👍

File: src/app.js [hunk 2]
```diff
@@ -59,12 +45,12 @@ class App extends React.Component {
   }
 
   render() {
-    const modalItem = this.state.items.find(item => item.id === this.state.modalTarget);
+    const modalItem = this.state.economy.getSidebarItems().find(item => item.id === this.state.modalTarget);      // TODO: WRONG USE OF SIDEBAR ITEMS, yOU RUBE
```
Comment (garrettgsb) [inline src/app.js]: `yOU` -> `YOU`

File: src/app.js [hunk 3]
```diff
@@ -124,18 +112,18 @@ class SidebarItem extends React.Component {
 
 class TheModal extends React.Component {
   render() {
-    const { item } = this.props;
-    if (item.modal) {
-      const Modal = React.createElement(item.modal);
+    const { eso } = this.props;
+    const ContainedView = eso.getView && eso.getView();
```
Comment (garrettgsb) [inline src/app.js]: Discussion, non-blocking: In the future, I imagine we'll want to fail early and loudly when something that doesn't have a `getView` is passed to `TheModal`. If not, then I suppose we'll want to extend what happens on line 124.

For now, this is a nice way to build ESOs incrementally without making the app unusable.

File: src/economy.js [hunk 4]
```diff
@@ -0,0 +1,74 @@
+
+import MineMezzo from './modules/mine/mezzo.js';
+
+import ResourcePool from './resourcePool.js';
+
+//      // Mock map objects
```
Comment (garrettgsb) [inline src/economy.js]: Double comments 😭

File: src/economy.js [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+    // TODO: maybe some consistent sorting pattern?
+    return this.esos;
+  }
+
+  getMapItems() {
+    // TODO: return whatever ESOs have a location?
+    return this.esos;
+  }
+
+  addNewDamnThing(thingName) {
```
Comment (garrettgsb) [inline src/economy.js]: Is this meant to be the method that we use to create a new ESO?

File: src/modules/mine/mezzo.js [hunk 6]
```diff
@@ -0,0 +1,26 @@
+import React from 'react';
+
+
+export default class MineMezzo {
+  constructor() {
```
Comment (garrettgsb) [inline src/modules/mine/mezzo.js]: y u empty ctor 🌚

File: src/modules/mine/mezzo.js [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+  getView() {
+    return MineGameView;
+  }
+
+}
+
+
+const MineGameView = () => {
+  return (
+    <div> look I am a respectable GameView for a respectable Mine </div>
```
Comment (garrettgsb) [inline src/modules/mine/mezzo.js]: Yes you are 👍

File: src/modules/mine/mezzo.js [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+  );
+
+}
+
+// class MineGameView extends React.Component {
```
Comment (garrettgsb) [inline src/modules/mine/mezzo.js]: Do you want this here or baleet?

File: src/resourcePool.js [hunk 9]
```diff
@@ -0,0 +1,75 @@
+
```
Comment (garrettgsb) [inline src/resourcePool.js]: How much of this is just moved, vs. actually modified? I'm not sure which parts to pay attention to.

File: src/app.js (hunk 2 above)
Comment (jholman) [reply src/app.js]: Gotta fix the whole line ASAP, I like the tpyo.

File: src/app.js (hunk 3 above)
Comment (jholman) [reply src/app.js]: Yeah, I had the same thought.  Either have faith that it's okay, or throw a tantrum, not this halfway thing.  But I couldn't make up my mind.

File: src/economy.js (hunk 5 above)
Comment (jholman) [reply src/economy.js]: I think it's one possible proposal for a possible way, not necessarily the correct way.

File: src/modules/mine/mezzo.js (hunk 8 above)
Comment (jholman) [reply src/modules/mine/mezzo.js]: Whole thing's gonna shift to another file, like `src/modules/mine/GameView.js`, real soon now.

File: src/resourcePool.js (hunk 9 above)
Comment (jholman) [reply src/resourcePool.js]: Look at the diff in the tests at the bottom first, then.  I ripped out all the clever.

**Pull request merged**